maestro-client/src/main/java/maestro/Maestro.kt

```diff
@@ -278,9 +292,12 @@ class Maestro(private val driver: Driver) : AutoCloseable {
         repeat(retries) {
             if (longPress) {
                 driver.longPress(Point(x, y))
-            } else {
-                driver.tap(Point(x, y))
-            }
+            } else if (tapRepeat != null) {
+                for (i in 0 until tapRepeat.repeat) {
+                    driver.tap(Point(x, y))
+                    if (tapRepeat.repeat > 1) Thread.sleep(tapRepeat.delay) // do not wait for single taps
+                }
+            } else driver.tap(Point(x, y))
```

Comment:
When tapRepeat is passed to the driver, we can better fit the specified delay.
In the current implementation, the roundtrip from driver to device is added to the delay for each tap.

Note that XCTest has api to send a sequence of taps with delays to the device, which allows for much-much more precise timing.
(see `XCPointerEventPath` in the xcuitest driver code)

Reply:
There's also `XCPointerEventPath.pressDownAtOffset(TimeInterval)` available in private API

Reply:
> In the current implementation, the roundtrip from driver to device is added to the delay for each tap.

I can subtract the execution delay from the overall delay. That would also be consistent with both platforms.
Tbh I don't think we need to complicate this any further by using other apis.